extract.go

```diff
@@ -30,10 +30,11 @@ import (
 var _typeOfIn = reflect.TypeOf(In{})
 
 // Extract fills the given struct with values from the dependency injection
-// container on application start.
+// container on application initialization. The target MUST be a pointer to a
+// struct. Only exported fields will be filled.
 //
-// The target MUST be a pointer to a struct. Only exported fields will be
-// filled.
+// Extract will be deprecated soon: use Populate instead, which doesn't
```

Comment:
Should we make the change necessary to deprecate Extract before adding these docs in? That way this can simply read: "Extract is deprecated: ..."

Reply:
I'd prefer not to. Internally, deprecating Extract will immediately break any build that's using the `staticcheck` linter, and I'd like to think more carefully about how to gracefully migrate those users without affecting stability over the holidays.

Basically, this is going to take some legwork, and I'd like to land these doc changes without blocking on that.